### ISSUE TITLE
[iOS] Unable to escape multi-block of a list item

### ISSUE THREAD

**Issue by niegowski** [ckeditor/ckeditor5#15677]
## 📝 Provide detailed reproduction steps (if any)

Create a list item with a single following empty block (inside the same list item). Press enter in the last empty block.

### ✔️ Expected result

The block is converted to a separate list item.

### ❌ Actual result

https://github.com/ckeditor/ckeditor5/assets/1232187/2a4741c5-9ad0-4187-9b7d-5cb4f6c96a26

### ❓ Possible solution

This might be related to beforeinput events, maybe it does not fire the 'enter' but rather 'soft enter'.

## 📃 Other details

* Browser: …
* OS: …
* First affected CKEditor version: …
* Installed CKEditor plugins: …

---

If you'd like to see this fixed sooner, add a 👍 reaction to this post.

**Comment by Witoso** [comment]
It it a regression?

**Comment by niegowski** [comment]
> It it a regression?

No. Marked is a potential blocker since now document lists are the default.

**Comment by Witoso** [comment]
Rel: #14540

**Comment by Mgsy** [comment]
Simpler scenario:

1.  Create a new list
2.  Type something in the first item
3.  Tap _Enter_
4.  Tap _Enter_ again to leave the list

In this case, the user is unable to leave the list
